packages/wire/lib/gossip/DnsPeerQuery.ts

```diff
@@ -0,0 +1,137 @@
+     * It follows the format of the address descriptor type specified in BOLT #7.
+     */
+    a?: number;
+
+    /**
+     * The bech32-encoded node id used to retrive the result of a single node.
+     */
+    l?: string;
+
+    /**
+     * The number of desired reply records.
+     */
+    n?: number;
+}
+
+class SrvRecord {
+    /**
+     * The hostname associated with the srv record.
+     */
+    name: string;
+
+    /**
+     * The port supported by the srv record.
+     */
+    port: number;
+
+    /**
+     * The priority of the srv record.
+     */
+    priority: number;
+
+    /**
+     * The weight of the srv record.
+     */
+    weight: number;
+}
+
+/**
+ * Performs a dns lookup on a dns seed to discover lightning nodes.
+ */
+export class DnsPeerQuery {
+    private resolver: any;
+    logger: ILogger;
+
+    constructor(logger: ILogger, resolver?: any) {
+        this.logger = logger;
+        this.resolver = resolver || new Resolver();
+    }
+
+    public async query(dnsPeerQueryOptions: DnsPeerQueryOptions): Promise<Array<PeerHostRecord>> {
+        let dnsSeed = this.buildUrl(dnsPeerQueryOptions);
+        let peerSrvRecords: Array<SrvRecord> = await this.getPeerSrvRecords(dnsSeed);
+
+        let peerHostRecordPromises = [];
+
+        for (let peerSrvRecord of peerSrvRecords) {
+            peerHostRecordPromises.push(this.createPeerHostRecord(peerSrvRecord));
+        }
+
+        let peers = await Promise.all(peerHostRecordPromises);
+        return peers;
+    }
+
+    private async createPeerHostRecord(peerSrvRecord) {
+        let peerAddress = await this.resolveSrvNameToIp(peerSrvRecord.name);
+        return new PeerHostRecord(
+            this.getPublicKeyFromSrvRecord(peerSrvRecord),
+            peerAddress,
+            peerSrvRecord.port,
+        );
+    }
+
+    private buildUrl(dnsPeerQueryOptions: DnsPeerQueryOptions) {
+        let { seedDomain: domain, r, a, l, n } = dnsPeerQueryOptions;
+
+        let dnsSeed = domain;
+
+        if (n) {
+            dnsSeed = `n${n}.${dnsSeed}`;
+        }
+
+        if (l) {
+            dnsSeed = `l${l}.${dnsSeed}`;
+        }
+
+        if (a) {
+            dnsSeed = `a${a}.${dnsSeed}`;
+        }
+
+        if (r) {
+            dnsSeed = `r${r}.${dnsSeed}`;
+        }
+
+        return dnsSeed;
+    }
+
+    private getPublicKeyFromSrvRecord(srvRecord: SrvRecord): Buffer {
+        let domainComponents = srvRecord.name.split(".");
+        let { words } = bech32.decode(domainComponents[0]);
+        return Buffer.from(bech32.fromWords(words));
+    }
+
+    private async getPeerSrvRecords(dnsSeed: string): Promise<Array<SrvRecord>> {
+        let results = [];
+
+        try {
+            results = await this.resolver.resolveSrv(dnsSeed);
+        } catch (e) {
+            this.logger?.error(e);
+        }
+
+        return results;
+    }
+
+    private async resolveSrvNameToIp(hostname: string): Promise<number> {
```

Comment:
I believe this returns a string

Reply:
Whoops, just changed signature to return a string